src/run.js

```diff
@@ -301,34 +315,76 @@ const processPage = ({
-        // for each, collect the URL of all the ones we're going to assess.
-        Array.from(document.querySelectorAll('link')).forEach(link => {
-          if (
-            link.href &&
-            (link.rel === 'stylesheet' ||
-              link.href.toLowerCase().endsWith('.css')) &&
-            !link.href.toLowerCase().startsWith('blob:') &&
-            link.media !== 'print' &&
-            !link.href.toLowerCase().startsWith('data:')
-          ) {
-            // Fragments are omitted from puppeteer's response.url(),
-            // so we need to strip them here, otherwise the hrefs
-            // won't always match when we check for missing ASTs.
-            hrefs.push(link.href.split('#')[0]);
+        const styles = [];
+        const isCssStyleTag = elem =>
+          elem.tagName === 'STYLE' &&
+          (!elem.type || elem.type.toLowerCase() === 'text/css');
+        const isStylesheetLink = elem =>
+          elem.tagName === 'LINK' &&
+          elem.href &&
+          (elem.rel.toLowerCase() === 'stylesheet' ||
+            elem.href.toLowerCase().endsWith('.css')) &&
+          !(
+            elem.href.toLowerCase().startsWith('data:') ||
+            elem.href.toLowerCase().startsWith('blob:') ||
+            elem.media.toLowerCase() === 'print'
+          );
+        // #fragments are omitted from puppeteer's response.url(), so
+        // we need to strip them from stylesheet links, otherwise the
+        // hrefs won't always match when we check for missing ASTs.
+        const defragment = href => href.split('#')[0];
+        const pageUrl = defragment(window.location.href);
+        // Create a unique identifier for each style tag by appending
+        // an xpath-like fragment to the page URL.  This allows us to
+        // preserve the relative ordering of external stylesheets and
+        // inline style tags.
+        const styleTagUri = () => `${pageUrl}#style[${styles.length}]`;
+        // Loop over all 'link' and 'style' elements in the document,
+        // in order of appearance. For each element, collect the URI
+        // of all the ones we're going to assess. For style elements,
+        // also extract each tag's content.
+        Array.from(document.querySelectorAll('link, style')).forEach(elem => {
+          if (isStylesheetLink(elem)) {
+            const href = defragment(elem.href);
+            hrefs.push(href);
+          } else if (isCssStyleTag(elem)) {
+            const href = styleTagUri();
+            const text = elem.innerHTML;
+            styles.push({ href, text });
+            hrefs.push(href);
           }
         });
-        return {
-          html: document.documentElement.outerHTML,
-          hrefs
-        };
+        return { html, hrefs, styles };
       });
 
       const htmlWithJavascript = evalWithJavascript.html;
       doms.push(cheerio.load(htmlWithJavascript));
+
+      if (options.styletags) {
```

Comment:
Can you move this up to where all the other `options` are dealt with. E.g.

```javascript
const styletags = options.styletags || false;

....

if (styletags) {
...
```

Reply:
Mainly for consistency and that it feels like we're done with `options` after it has been turned into variables.

Reply:
Done.